src/Workspaces/Core/Portable/LanguageServices/DeclaredSymbolFactoryService/AbstractDeclaredSymbolInfoFactoryService.cs

```diff
@@ -31,6 +44,33 @@ internal abstract class AbstractDeclaredSymbolInfoFactoryService : IDeclaredSymb
         private static readonly ObjectPool<Dictionary<string, string>> s_aliasMapPool
             = SharedPools.StringIgnoreCaseDictionary<string>();
 
+        protected AbstractDeclaredSymbolInfoFactoryService()
+        {
+        }
+
+        protected abstract SyntaxList<TMemberDeclarationSyntax> GetChildren(TCompilationUnitSyntax node);
+        protected abstract SyntaxList<TMemberDeclarationSyntax> GetChildren(TNamespaceDeclarationSyntax node);
+        protected abstract SyntaxList<TMemberDeclarationSyntax> GetChildren(TTypeDeclarationSyntax node);
+        protected abstract IEnumerable<TMemberDeclarationSyntax> GetChildren(TEnumDeclarationSyntax node);
```

Comment:
enums are a SeparatedSyntaxList in C#, and a SyntaxList in VB.  so we need an IEnumerable here.

Reply:
i considered making this generic and adding `TEnumMemberDeclarationListSyntax ... where TEnumMemberDeclarationListSyntax  : IEnumerable<TMemberDeclarationSyntax>`.  However, that doesn't help as foreaching will still allocate the IEnumerator.  We'd need to have abstracts for the count/indexing to avoid any allocations here, and it just seemed to overwrought to be worth it.